supplementary/alica_ros1/alica_tests/src/test/test_alica_simple_plan.cpp

```diff
@@ -1,79 +1,37 @@
-namespace
-{
-
-class AlicaSimplePlan : public AlicaTestFixture
+namespace alica::test
 {
-protected:
-    const char* getRoleSetName() const override { return "Roleset"; }
-    const char* getMasterPlanName() const override { return "SimpleTestPlan"; }
-    bool stepEngine() const override { return false; }
-};
-
 /**
  * Tests whether it is possible to run a behaviour in a primitive plan.
  */
-TEST_F(AlicaSimplePlan, runBehaviourInSimplePlan)
+TEST_F(TestFixture, runBehaviourInSimplePlan)
 {
-    ASSERT_NO_SIGNAL
-    ae->start();
-    alica::AlicaTime sleepTime = alica::AlicaTime::seconds(1);
-    uint8_t timeoutCount = 0;
-
-    do {
-        ae->getAlicaClock().sleep(sleepTime);
-    } while (!alica::test::Util::isPlanActive(ae, 1412252439925));
-
-    // Check whether RC can be called
-    EXPECT_TRUE(ae->getPlanBase().getRootNode()->isRuntimeConditionValid());
-    // Check whether RC has been called
-    EXPECT_GE(CounterClass::called, 1);
-
-    while (!alica::test::Util::isStateActive(ae, 1412761855746) && timeoutCount < 5) {
-        ae->getAlicaClock().sleep(sleepTime);
-        timeoutCount++;
-    }
-    timeoutCount = 0;
-
-    // Check final state
-    EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1412761855746));
-    // Check execution of final state behaviour
-    EXPECT_TRUE(alica::test::Util::isPlanActive(ae, 1402488848841));
-
-    // We assume at least 30 calls to Attack in (3 * sleepTime) seconds.
-    while (dynamic_cast<alica::Attack*>(alica::test::Util::getBasicBehaviour(ae, 1402488848841, 0))->callCounter < 30 && timeoutCount < 3) {
-        ae->getAlicaClock().sleep(sleepTime);
-        timeoutCount++;
-    }
-    timeoutCount = 0;
-
-    EXPECT_GE(dynamic_cast<alica::Attack*>(alica::test::Util::getBasicBehaviour(ae, 1402488848841, 0))->callCounter, 30);
-    EXPECT_GT(dynamic_cast<alica::Attack*>(alica::test::Util::getBasicBehaviour(ae, 1402488848841, 0))->initCounter, 0);
-    CounterClass::called = 0;
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "SimpleTestPlanState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("SimpleTestPlan"));
+    ASSERT_NE(nullptr, _tc->getActivePlan("SimpleTestPlan")) << _tc->getLastFailure();
+
+    // TestState1
+    ASSERT_TRUE(_tc->isStateActive("SimpleTestPlan", "TestState1")) << _tc->getLastFailure();
+    ASSERT_NE(nullptr, _tc->getActiveBehaviour("MidFieldStandard")) << _tc->getLastFailure();
+    ASSERT_EQ(nullptr, _tc->getActiveBehaviour("Attack")) << _tc->getLastFailure();
+
+    // TestState2
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->isStateActive("SimpleTestPlan", "TestState2"));
+    ASSERT_EQ(_tc->getActivePlan("SimpleTestPlan")->getName(), "SimpleTestPlan") << _tc->getLastFailure();
+    ASSERT_NE(nullptr, _tc->getActiveBehaviour("Attack")) << _tc->getLastFailure();
+    ASSERT_EQ(nullptr, _tc->getActiveBehaviour("MidFieldStandard")) << _tc->getLastFailure();
```

Comment:
I'd rather not use Attack or MidFieldStandard behaviour's here because they don't do anything close to what you would expect from their name. Eg. MidFieldStandard is really SucceedAfterNRunIterations